teps/0101-env-in-pod-template.md

```diff
@@ -0,0 +1,642 @@
+if this problem is solved?).
+-->
+
+1. In the first case, common environment variables can be defined in a single place on a `PipelineRun` level.
+   Values can be specified as literals or through Kubernetes references.
+   Variables defined on a `PipelineRun` or `TaskRun` level are then available in all steps.
+   That allows to simplify the Tekton run resource configuration and significantly reduce the size of
+   `PipelineRun` and `TaskRun` resource,
+   by excluding the common environment variables like: static global settings, common values coming from metadata, ...
+
+2. Secondly, environment variables defined in steps can be easily overwritten by the ones from `PipelineRun` and `TaskRun`.
+  With that, common settings like API keys, connection details, ... can be optionally overwritten in a single place.
+  
+3. For Cloud Providers, it's very common to inject user credentials using Kubernetes API `valueFrom` to avoid credentials
+   being exposed to the PodSpec. Since each cloud provider has different credential format, able to assign environment 
+   variables at the `PipelineRun` and `TaskRun` can reuse the same task with different Cloud Provider credentials.
+   Kubernetes API `valueFrom` can also refe to values in the pod labels/annotations for specific Kubernetes cluster
+   information such as namespace, application labels, and service annotations.
+
+## Requirements
+
+<!--
+Describe constraints on the solution that must be met. Examples might include
+performance characteristics that must be met, specific edge cases that must
+be handled, or user scenarios that will be affected and must be accomodated.
+-->
+
+1. In the first case, common environment variables can be defined in a single place on a `PipelineRun` level.
+   Values can be specified as literals or through references.
+   Variables defined on a `PipelineRun` or `TaskRun` level are then available in all steps.
+   That allows to significantly reduce the size of `PipelineRun` and `TaskRun` resource,
+   by excluding the common environment variables like: static global settings, common values coming from metadata, ...
+
+2. Secondly, environment variables defined in steps can be easily overwritten by the ones from `PipelineRun` and `TaskRun`.
+  With that, common settings like API keys, connection details, ... can be optionally overwritten in a single place. For
+  example, if both `PipelineRun` and task `StepTemplate` has environment variables PROXY, it will overwrite the task
+  `StepTemplate` with the environment variable values inside `PipelineRun`.
+  
+3. Allow cluster admin to define a list of cluster-wide forbidden environment variables in the Tekton `config-defaults` 
+   configmap. When users define environment variables in the Taskrun and Pipelinerun spec, check the list of forbidden
+   environment variables and throw out a validation error if any of the environment variables is forbidden. 
+
```

Comment:
Also can be specify a global level default pod template envs like here https://github.com/tektoncd/pipeline/blob/main/config/config-defaults.yaml#L57 so the ones which i want to get added for very pipelinerun/taskrun can be specified here